tests/integration/test_charm.py

```diff
@@ -396,12 +501,14 @@ async def test_seldon_predictor_server(ops_test: OpsTest, server_config, url, re
     stop=tenacity.stop_after_attempt(60),
     reraise=True,
 )
-async def assert_removed(ops_test: OpsTest):
+async def assert_application_removed(ops_test: OpsTest):
     """Remove application and verify it is removed."""
-    logger.info(f"Removing application {APP_NAME}")
+    logger.info(f"Removing application {APP_NAME} from model {ops_test.model_name}")
     # TO-DO: use this: await ops_test.run("juju", "remove-application", f"{APP_NAME}")
-    subprocess.run(["juju", "remove-application", f"{APP_NAME}"])
-    assert APP_NAME not in ops_test.model.applications
+    subprocess.check_output(
+        ["juju", "remove-application", "-m", f"{ops_test.model_name}", f"{APP_NAME}"]
+    )
```

Comment:
why do we need subprocess instead of `await ops_test.model.remove_application(APP_NAME, block_until_done=True)`

Reply:
Call to `await ops_test.model.remove_application(APP_NAME, block_until_done=True)` hangs indefintely and does not delete application fully. It is stuck in `terminated` state.

Reply:
Yes, sidecar.
Controller 2.9.43
Juju 2.9.43

Reply:
terminated thing should be actually fixed

please remove this and replace with 2.9/stable channel
https://github.com/canonical/seldon-core-operator/blob/58a245b049ea907eb8b05a924e9de98aed2f5e1b/.github/workflows/integrate.yaml#L59-L60

then it should work with `await ops_test.model.remove_application(APP_NAME, block_until_done=True)`

Reply:
Works!